### PR TITLE
Add a simple MinGW example

### PR DESCRIPTION
**Pull request by abbec**
This has an artificial dependency on `pthreads` (a commonly needed one)
combined with a `build.rs` to trigger cross-compilation in combination
with compiling for the build platform.

### CODE REVIEW

File: examples/cross-mingw/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/target
```
Comment (oxalica) [inline examples/cross-mingw/.gitignore:1]: `target` is already ignored in root `.gitignore`. No need to duplicate.

File: examples/cross-mingw/build.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (oxalica) [inline examples/cross-mingw/build.rs:1]: It's better to put a link to #59 here.

File: examples/cross-mingw/shell.nix [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# See docs/cross_compilation.md for details.
+(import <nixpkgs> {
+  crossSystem = {
+    config = "x86_64-w64-mingw32";
+  };
```
Comment (oxalica) [inline examples/cross-mingw/shell.nix:5]:
```suggestion
  crossSystem = "x86_64-w64-mingw32";
```
Simple config can be just written as a string.

File: examples/cross-mingw/shell.nix [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+}).callPackage (
+{ mkShell, stdenv, rust-bin, windows }:
+mkShell {
+  nativeBuildInputs = [
+    rust-bin.stable.latest.minimal
+  ];
+
+  buildInputs = [ windows.pthreads ];
+
+  CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER = "${stdenv.cc.targetPrefix}cc";
```
Comment (oxalica) [inline examples/cross-mingw/shell.nix:16]: Maybe add `wine64` as the test runner? You can check aarch64 example for how to specify it.
https://github.com/oxalica/rust-overlay/blob/db229315e3d220ce90eec3db8fefc944a007be74/examples/cross-aarch64/shell.nix#L13-L18

**Review by oxalica — CHANGES_REQUESTED** [review]
The example should be able to build now. Let's fine tune it.

File: examples/cross-mingw/shell.nix (hunk 3 above)
Comment (abbec) [reply examples/cross-mingw/shell.nix:5]: Actually doing so seemed to break the splicing? If I do the suggested change, it tries to use `mkShell` for `x86_64-w64-mingw32`.

File: examples/cross-mingw/shell.nix (hunk 4 above)
Comment (abbec) [reply examples/cross-mingw/shell.nix:17]: Seemed to work ok. Wine is a bit chatty and has some side effects when running it for the first time but if we are ok with that this should be good to go.

File: examples/cross-mingw/shell.nix (hunk 3 above)
Comment (oxalica) [reply examples/cross-mingw/shell.nix:5]: Don't you just want `x86_64-w64-mingw32`? I tested it and it also works fine.

Comment (oxalica) [reply examples/cross-mingw/shell.nix:5]: Oh it actually doesn't work. My bad.

**Review by oxalica — APPROVED** [review]

**Pull request merged**